src/components/sidebar/Sidebar.tsx

```diff
@@ -16,13 +16,68 @@ import {
 import { CloseIcon, EditIcon, SearchIcon } from '@chakra-ui/icons'
 import DragItem from './DragItem'
 import { menuItems, MenuItem } from '~componentsList'
+import { useSelector } from 'react-redux'
 import {
-  cmenuItems,
-  CMenuItem,
-} from '../../custom-components/customComponentsList'
+  getCustomComponents,
+  getSelectedCustomComponentId,
+} from '~core/selectors/customComponents'
+import useDispatch from '~hooks/useDispatch'
+import API from '~custom-components/api'
 
 const Menu = () => {
   const [searchTerm, setSearchTerm] = useState('')
+  const dispatch = useDispatch()
+  const currentComponents = useSelector(getCustomComponents)
+  const selectedComponent = useSelector(getSelectedCustomComponentId)
+
+  const getObjectDiff = (updatedList: Record<string, unknown>) => {
+    let deletedComponents = Object.keys(currentComponents).filter(
+      component => !Object.keys(updatedList).includes(component),
+    )
+    let newComponents = Object.keys(updatedList).filter(
+      component => !Object.keys(currentComponents).includes(component),
+    )
+    return {
+      deletedComponents: deletedComponents,
+      newComponents: newComponents,
+    }
+  }
+
+  useEffect(() => {
+    const interval = setInterval(async () => {
+      const newComponentsList = await API.get('/refresh').then(res => res.data)
+      const componentDiffs = getObjectDiff(newComponentsList)
+      if (componentDiffs.deletedComponents.length) {
```

Comment:
No need of if condition

Reply:
we added the if conditions because there are many cases when there won't be any new component or deleted component. If we don't do that, won't it be calling the api every 3 seconds?

Reply:
no it will not call the apis. If `componentDiffs.deletedComponents` is empty, map function will not be called

Reply:
okay, done